### PR TITLE
Added s3 file upload

### PR DESCRIPTION
**Pull request by jbevan4**
**What**  
Allows files upload by exposing the url and fields needed

**Why**  
So we can upload files

**Anything else?**
NA

### CODE REVIEW

File: lib/usecases/getDropBox.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export default ({ gateways: { dropbox: dropboxes, document: documents } }) => {
```
Comment (bjpirt) [inline lib/usecases/getDropBox.js:1]: Not sure we need this file

File: lib/gateways/s3.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      Key: key,
+      Expires: 60
+    });
+  }
+
+  /**
+   * Fetches all documents that have been uploaded to a given dropbox.
+   * @param {String} dropboxId the id of the dropbox
+   */
+  function getByDropboxId(dropboxId) {
```
Comment (bjpirt) [inline lib/gateways/s3.js:22]: I think we can remove this function

File: lib/usecases/getSecureUploadUrl.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { nanoid } from 'nanoid';
+import createGateway from '../gateways/s3';
+import s3config from '../s3config';
+
+const documents = createGateway({
+  ...s3config,
+  configuration: { urlPrefix: process.env.URL_PREFIX, maxUploadBytes: 20971520 }
+});
+
+export default async dropboxId => {
```
Comment (neilkidd) [inline lib/usecases/getSecureUploadUrl.js:10]: As per @bjpirt 's comments. The dropbox code can be deleted.

File: lib/gateways/s3.js (hunk 2 above)
Comment (StephenThomson) [reply lib/gateways/s3.js:22]: Whether or not we need this is part of how we store the documents discussion

File: lib/usecases/getSecureUploadUrl.js (hunk 3 above)
Comment (StephenThomson) [reply lib/usecases/getSecureUploadUrl.js:10]: The dropboxId in this context is just the name of the folder in s3. There is still ongoing discussion on how to use folders.

Comment (jbevan4) [reply lib/usecases/getSecureUploadUrl.js:10]: This is the code which generates the secureurl

File: lib/gateways/s3.js (hunk 2 above)
Comment (StephenThomson) [reply lib/gateways/s3.js:22]: Based on the previous discussion this can now be removed.

File: lib/usecases/getDropBox.js (hunk 1 above)
Comment (StephenThomson) [reply lib/usecases/getDropBox.js:1]: Agreed

**Pull request merged**